server/tck/src/main/java/org/eclipse/microprofile/graphql/tck/apps/basic/api/SourceTestApi.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.graphql.tck.apps.basic.api;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import org.eclipse.microprofile.graphql.DateFormat;
+import org.eclipse.microprofile.graphql.DefaultValue;
+import org.eclipse.microprofile.graphql.GraphQLApi;
+import org.eclipse.microprofile.graphql.Query;
+import org.eclipse.microprofile.graphql.Source;
+
+/**
+ * {@code @Source} testing.
+ */
+@GraphQLApi
+public class SourceTestApi {
+
+    @Query
+    public SourceType getSource() {
+        return new SourceType();
+    }
+
+    public String stringInput(@Source SourceType source, String input) {
```

Comment:
The "null" test made me think that it might be nice to have another test case for when the `@NonNull` annotation is used - for example:
```
public String nonNullStringInput(@Source SourceType source, @NonNull String input) { //...
```

then the test input would look the same as the current null test, but the output would be an error.  Wdyt?

Reply:
Thanks for the suggestion, added tests with `@NonNull` and `@Name`!